File: FormalBench-data/diverse/rule_23/SqrtRoot.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class SqrtRoot {

	public static int sqrtRoot(int num) {
		if (0 > num) {
			return -1;
		}
		int left = 0;
		int right = num;
		while (right >= left) {
			int mid = left + (right - left) / 2;
			if (mid * mid == num) {
				return mid;
			} else if (num > mid * mid) {
				left = mid + 1;
			} else {
				right = mid - 1;
			}
		}
		return right;
	}
}

File: FormalBench-data/FilteredData/fault_mutants/run_ukcw__uc/SqrtRoot/mutants/14/c/SqrtRoot.c
#include "java_arrays.h"
#include <stdint.h>

static int32_t java_add(int32_t left, int32_t right) { return (int32_t)((uint32_t)left + (uint32_t)right); }
static int32_t java_sub(int32_t left, int32_t right) { return (int32_t)((uint32_t)left - (uint32_t)right); }
static int32_t java_mul(int32_t left, int32_t right) { return (int32_t)((uint32_t)left * (uint32_t)right); }
static int32_t java_neg(int32_t value) { return (int32_t)(0u - (uint32_t)value); }

int32_t sqrtRoot(int32_t num) {
    if ((num < INT32_C(0))) {
        return java_neg(INT32_C(1));
    }
    int32_t left = INT32_C(0);
    int32_t right = num;
    while ((left <= right)) {
        int32_t mid = java_add(java_add(left, java_sub(right, left)), INT32_C(2));
        if ((java_mul(mid, mid) == num)) {
            return mid;
        } else {
            if ((java_mul(mid, mid) < num)) {
                left = java_add(mid, INT32_C(1));
            } else {
                right = java_sub(mid, INT32_C(1));
            }
        }
    }
    return right;
}
